code/API_definitions/qod-api.yaml

```diff
@@ -328,6 +477,166 @@ components:
           - 5070
       description: |
         Ports may be specified as a list of ranges or single ports.
+    QoSProfileId:
+      type: string
+      example: 123e4567-e89b-12d3-a456-426614174000
+      description: |
+        Unique ID of the QoS Profile to be used for this session.
+      minLength: 36
+      maxLength: 36
+      format: string
```

Comment:
Agree, should be `format: uuid`.

Reply:
As commented above, I have doubts about the convenience of having UUID in all cases. If this is to be passed when creating a request, it would be always necessary to discover profiles, to know their UUIDs. It is very related with the other topic about the coexistence of legacy and new profiles.

Reply:
If SCEF / NEF is the network element implementing the QoS profile, then my expectation is that the `id` parameter would be the one passed as `qosReference` to the SCEF / NEF. From that point of view, constraining `id` / `qosReference` to be a uuid would greatly simplify data validation, and I would prefer to avoid having to do any mapping between `id` and some other set of `qosReference` values.

However, I take the point that this is linked to the discussion about the use of `name`, and whether that should be unique, and hence the QoS profile key. `name` should not be constrained to be a uuid. So I'd say if `name` is made the key, but we keep `id` as an optional QoS profile parameter, then we just need to say `id` is a string, and it can be set to be a uuid for each QoS profile if that is convenient.

But I would prefer to avoid passing `name` to the SCEF / NEF as `qosReference`, as `name` should mean something to the API consumer that will not necessarily coincide with the internal naming conventions for `qosReference`. I would expect QoS profiles to have an associated `id`, whether that is formally part of the definition, or only an internal parameter.